### ISSUE TITLE
[Bug]: Unable to build Mapbox Navigation SDK from `release-v2.4` branch using Xcode `13.2.1`.

### ISSUE THREAD

**Issue by MaximAlien** [mapbox/mapbox-navigation-ios#4047]
### Mapbox Navigation SDK version

`release-v2.4`

### Steps to reproduce

Run `$ swift package update` in the root directory.

### Expected behavior

Project should update dependencies without any issues.

### Actual behavior

Such error occurs:
```shell
error:   Dependencies could not be resolved because root depends on 'mapbox-navigation-native-ios' 94.0.3..<95.0.0 and root depends on 'mapbox-maps-ios' 10.4.3..<11.0.0.
  'mapbox-navigation-native-ios' is incompatible with 'mapbox-maps-ios' because 'mapbox-navigation-native-ios' 94.0.3..<95.0.0 practically depends on 'mapbox-common-ios' 21.2.0..<22.0.0 (1).
  'mapbox-maps-ios' >= 10.4.3 practically depends on 'mapbox-common-ios' {21.2.1, 21.3.0, 22.0.0, 22.1.0} because 'mapbox-maps-ios' 10.6.0..<10.7.0-beta.1 depends on 'mapbox-common-ios' 22.0.0 and 'mapbox-maps-ios' >= 10.7.0 depends on 'mapbox-common-ios' 22.1.0.
  'mapbox-maps-ios' {10.4.3..<10.6.0, 10.7.0-beta.1..<10.7.0} practically depends on 'mapbox-common-ios' {21.2.1, 21.3.0} because 'mapbox-maps-ios' 10.5.0..<10.6.0-beta.1 depends on 'mapbox-common-ios' 21.3.0.
  'mapbox-maps-ios' {10.4.3..<10.5.0, 10.6.0-beta.1..<10.6.0, 10.7.0-beta.1..<10.7.0} practically depends on 'mapbox-common-ios' 21.2.1 because no versions of 'mapbox-maps-ios' match the requirement {10.5.0-beta.1..<10.5.0, 10.6.0-beta.1..<10.6.0, 10.7.0-beta.1..<10.7.0} and 'mapbox-maps-ios' 10.4.3..<10.5.0-beta.1 depends on 'mapbox-common-ios' 21.2.1.
  'mapbox-navigation-native-ios' 94.0.3..<94.0.5 practically depends on 'mapbox-common-ios' 21.2.0..<22.0.0 because 'mapbox-navigation-native-ios' 94.0.4..<94.0.5 depends on 'mapbox-common-ios' 21.2.0..<22.0.0 and 'mapbox-navigation-native-ios' 94.0.3..<94.0.4 depends on 'mapbox-common-ios' 21.2.1..<22.0.0.
     (1) As a result, 'mapbox-navigation-native-ios' 94.0.3..<95.0.0 practically depends on 'mapbox-common-ios' 21.2.0..<22.0.0 because 'mapbox-navigation-native-ios' 94.0.5..<95.0.0 depends on 'mapbox-common-ios' 21.2.1..<22.0.0.
```

### Is this a one-time issue or a repeatable issue?

repeatable

**Comment by MaximAlien** [comment]
Some additional details: based on logs it looks like Xcode `v13.2.1` fails to resolve Mapbox Common and wants to use `v21.4.0` instead of `v21.3.0` that is common for both Maps and Navigation Native. Here're logs when running:
```shell
$ /Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift package update --verbose
```
[output_xcode_13.2.1.txt](https://github.com/mapbox/mapbox-navigation-ios/files/9263344/output_xcode_13.2.1.txt)

On the other hand Xcode `v13.4.1` resolves Mapbox Common to `v21.3.0` that is common for both Maps and Navigation Native:

[output_xcode_13.4.1.txt](https://github.com/mapbox/mapbox-navigation-ios/files/9263348/output_xcode_13.4.1.txt)

**Comment by 1ec5** [comment]
Fixed in #4048.
